test/test_parse_es.py

```diff
@@ -84,6 +86,134 @@ def test_extract_number_es(self):
+    def test_bugged_output_wastebasket(self):
+        self.assertEqual(extract_datetime(
+            "11 jun", lang='es', anchorDate=datetime(1998, 6, 1))[0],
+            datetime(1998, 6, 11))
+        self.assertEqual(extract_datetime(
+            "11 junio", lang='es', anchorDate=datetime(1998, 6, 1))[0],
+            datetime(1998, 6, 11))
+        self.assertEqual(extract_datetime(
+            "11 jul", lang='es', anchorDate=datetime(1998, 7, 1))[0],
+            datetime(1998, 7, 11))
+        self.assertEqual(extract_datetime(
+            "11 ago", lang='es', anchorDate=datetime(1998, 8, 1))[0],
+            datetime(1998, 8, 11))
+        self.assertEqual(extract_datetime(
+            "11 sep", lang='es', anchorDate=datetime(1998, 9, 1))[0],
+            datetime(1998, 9, 11))
+
+        # It's also failing on years
+        self.assertEqual(extract_datetime(
+            "11 ago 1998", lang='es')[0], datetime(1998, 8, 11))
+
+    def test_extract_datetime_relative(self):
+        self.assertEqual(extract_datetime("esta noche", anchorDate=datetime(
+            1998, 1, 1), lang='es'),
+            [datetime(1998, 1, 1, 21, 0, 0), 'esta'])
+        self.assertEqual(extract_datetime(
+            "ayer noche", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 31, 21))
+        self.assertEqual(extract_datetime(
+            "el noche anteayer", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 30, 21))
+        self.assertEqual(extract_datetime(
+            "el noche ante ante ayer", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 29, 21))
+        self.assertEqual(extract_datetime(
+            "mañana en la mañana", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1998, 1, 2, 8))
+        self.assertEqual(extract_datetime(
+            "ayer en la tarde", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 31, 15))
+
+        self.assertEqual(extract_datetime(
+            "que año es", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1998, 1, 1))
+
+        self.assertEqual(extract_datetime("hoy 2 a.m.", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 2))
+        self.assertEqual(extract_datetime("hoy 2 am", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 2))
+        self.assertEqual(extract_datetime("hoy 2 p.m.", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 14))
+        self.assertEqual(extract_datetime("hoy 2 pm", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 14))
+
```

Comment:
In spanish we don't say "am" or "pm" but "de la madrugada", "de la mañana", "de la tarde", or "de la noche". The time frames varies from region to region, but usually are those:
- from 00:01 to 11:59 --> "de la mañana"
- from 12:01 to 20:00 --> "de la tarde"
- from 20:01 to 23:59 --> "de la noche"

There are some special time frames:
- 00:00 is "media noche" or "medianoche" and 12:00 is "medio día" or "mediodía"
- the time frame from 01:00 to 05:59 is usually also called "de la madrugada"

Reply:
This has been updated now.